ert_gui/main.py

```diff
@@ -10,29 +14,62 @@ def valid_file(fname):
     return fname
 
 
-def runExec(executable, args):
-    os.execvp(executable, [executable] + args)
+def valid_realizations(user_input):
+    validator = RangeStringArgument()
+    validated = validator.validate(user_input)
+    if validated.failed():
+        raise ArgumentTypeError(
+            "Defined realizations is not of correct format: {}".format(user_input))
+    return user_input
 
 
-def runGui(args):
-    runExec("python", ["-m", "ert_gui.gert_main"] + [args.config])
+def valid_weights(user_input):
+    validator = NumberListStringArgument()
+    validated = validator.validate(user_input)
+    if validated.failed():
+        raise ArgumentTypeError(
+            "Defined weights is not of correct format: {}".format(user_input))
+
+    return user_input
 
 
-def runCli(args):
-    runExec("python", ["-m", "ert_gui.cli"] +
-            [args.config, args.mode, args.target_case])
+def valid_name_format(user_input):
+    validator = ProperNameFormatArgument()
+    validated = validator.validate(user_input)
+    if validated.failed():
+        raise ArgumentTypeError(
+            "Defined name is not of correct format: {}".format(user_input))
+    return user_input
 
 
-def validate_cli_args(parser, args):
-    if args.mode == "ensemble_smoother" and args.target_case == "default":
+def valid_name_format_not_default(user_input):
+    if user_input == 'default':
         msg = "Target file system and source file system can not be the same. "\
-              "They were both: <default>. Please set using --target-case on "\
-              "the command line."
-        parser.error(msg)
+              "They were both: <default>."
+        raise ArgumentTypeError(msg)
+    valid_name_format(user_input)
+    return user_input
 
 
-def ert_parser():
-    parser = ArgumentParser(description="ERT - Ensemble Reservoir Tool")
+def range_limited_int(user_input):
+    try:
+        i = int(user_input)
+    except ValueError:
+        raise ArgumentTypeError("Must be a int")
+    if 0 < i < 100:
+        return i
+    raise ArgumentTypeError("Range must be in range 1 - 99")
+
+
+def runExec(executable, args):
+    os.execvp(executable, [executable] + args)
+
+
+def runGui(args):
+    runExec("python", ["-m", "ert_gui.gert_main"] + [args.config])
+
+
+def ert_parser(parser, args):
```

Comment:
I know I have arleady asked about this, but I am not sure I fully understood. Why do we need to pass in the parser? can't we just pass in the `args`?

Reply:
It's due to the testing and how i doing it.... When testing - if the parser is created inside ert_parser - a call to sys.args[0] i failing inside ArgumentParser class 
```
if prog is None:
            prog = _os.path.basename(_sys.argv[0])
```
Could be other options of doing this - but this is how far i've gotten and the easiest way to test for me i think! Previously we used mocks - but i think it was a bit messy....